src/types.ts

```diff
@@ -157,13 +160,29 @@ export interface Service {
 	user?: string;
 	userns_mode?: string;
 
-	volumes?: ListOfUniqueItems<string>;
+	volumes?: ListOfUniqueItems<string | VolumeRef>;
```

Comment:
This is a breaking change. Eg. the Supervisor will probably choke if it encounters `VolumeRef`.

Reply:
yeah. as I was wrote in the pr description, `VolumeRef` becomes the new canonical form for `service.volumes`... downstream would have to make the appropriate changes to support that

Reply:
Can you briefly explain how you envision that would work for us? What if I'm a user that has a recent CLI that includes these changes -- what should happen when I next push a release?

Reply:
I'm not sure if the CLI needs to deal with this field at all? but for the SV see my comment here: https://github.com/balena-io-modules/resin-compose-parse/pull/59#discussion_r690405993

Reply:
What I'm trying to get at is this:

This module is used to transform input into a canonical representation that the rest of the product understands. At this moment, this representation is for the most part identical to schema v2.1. A composition defines build-time and runtime configuration. Our product is in a somewhat unique position where runtime is far detached from build-time.

Making a breaking change to the build-time part of this representation merely requires clients to update -- that's fine. Making a breaking change to the runtime part however (eg. changing the type of `volumes`) is not trivial: existing devices have no way to cope with the changed runtime config and they'll just break if they encounter it.

So, at *some* point before this config reaches a device we need to transform it into something it already knows how to handle. What I'm asking is how do you plan to push the breaking change forward without breaking existing devices?

Reply:
BTW, if it helps, we're not wedded to any specific compose schema -- the whole point for this module is that we decouple user input from the actual data the rest of the platform understands. In other words, I'm personally not at all opposed to deviating from compose schema and adding our own attributes to it so we can introduce changes in a backwards-compatible way.

Reply:
One last point -- the reason I suggested in our previous discussion to go for the minor 2.3 increment instead of the v3 directly, is because I expected it could be accomplished without having to introduce breaking changes which would have avoided the need to figure all this out right now :)

Reply:
`target` is really only a build-time thing, so no need for the SV to require any changes. I only did the `VolumeRef` thing because it's part of 2.3 and I didn't know of the implications that run-time config changes would have

I agree then that we should probably only accept the long syntax on the input but normalize to the list of strings. which would make it backwards compatible